File: src/m/miscellaneous/isnans.py
import numpy as np


def isnans(array):
    """
    ISNANS: figure out if an array is nan. wrapper to isnan from matlab which stupidly does not allow this test  for structures!

       Usage:    isnans(array)

          See also : ISNAN
    """

    if isinstance(array, (tuple, list, dict)):
        returnvalue = 0
    else:
        returnvalue = np.isnan(array)

    return returnvalue

File: src/m/miscellaneous/isnans.m
function returnvalue=isnans(array)
%ISNANS - figure out if an array is nan; wrapper to isnan from MATLAB, which does not allow this test for structures
%
%   Usage:
%      returnvalue=isnans(array);
%
%   See Also:
%      ISNAN

if isstruct(array) 
	returnvalue=0;
elseif iscell(array)
	returnvalue=0;
else
	returnvalue=isnan(array);
end
